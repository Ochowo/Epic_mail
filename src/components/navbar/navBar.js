
import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faUserCircle } from '@fortawesome/free-solid-svg-icons';

import Logo from '../logo/logo';
import { logoutUser } from '../../actions/authAction';

class Header extends React.Component {
  onLogoutClick(e) {
    e.preventDefault();
    // eslint-disable-next-line react/destructuring-assignment
    this.props.logoutUser();
  }

  render() {
    const { auth } = this.props;
    const { isAuthenticated } = auth;

    const guestLink = (
      <header className="header uyu">
        <nav className="nav">
          <span className="logo lo">
            <Link to="/https://epic-mail04.herokuapp.com">
              <Logo />
            </Link>
          </span>
          <span className="help">
            <Link to="/inbox">Need help?</Link>
          </span>
        </nav>
      </header>
    );

    const authLink = (
      <header className="header header2 header4">
        <nav className="nav">
          <div className="flexboxx">
            <div className="logo bak">
              <Link to="/https://epic-mail04.herokuapp.com">
                <Logo />
              </Link>
              {' '}
            </div>
            <div className="search-container">
              <form action="#">
                <input type="text" placeholder="Search inbox.." name="search" className="bor" />
                <button type="submit" className="top-subb">
                  <i className="fa fa-search top-sub" />
                </button>
              </form>
            </div>
            <div className="userp">
              <FontAwesomeIcon icon={faUserCircle} className="top-us" />
              <div className="usern">username</div>
            </div>

            <button className="logout" type="submit" onClick={this.onLogoutClick.bind(this)}>
                logout
            </button>
          </div>
        </nav>
      </header>
    );
    return <div>{isAuthenticated ? authLink : guestLink}</div>;
  }
}
Header.propTypes = {
  logoutUser: PropTypes.func.isRequired,
  auth: PropTypes.shape({
    root: PropTypes.string,
    isAuthenticated: PropTypes.bool,
  }),
};
const mapStateToProps = state => ({
  auth: state.auth,
});
export default connect(
  mapStateToProps,
  { logoutUser },
)(Header);
export { Header };
