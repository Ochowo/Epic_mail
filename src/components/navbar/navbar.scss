@import '../../../public/style/colors';
@mixin logo {
    width: 15%;    
}
.header{
    @include logo;
    height: 70px;
    height:70px;
    padding: 20px 65px ;
    width: 100%;
    .lo {
        padding: 0;
        margin: 0;
      }
      .help {
        float: right;
        color: $font_color_2;
      }
}
.header2 {
    border-bottom: 1px solid #ededed;
}

.header4 {
    height: 70px;
    padding: 0 40px ;
    width: 100%;
    .flexboxx {
        @include logo;
        color: #707070;
        width: 100%;
        display: flex;
        flex-wrap: nowrap;
        .search-container {
            float: right;
            font-size: 17px;
            cursor: pointer;
            width: 50%;
            margin-top:1.2%;
            margin-left: 2%;
            .bor {
                border: 1px solid #bababa;
                width: 80%;
                padding: 1% 1%;
            }
            .top-subb {
                background-color: #fff;
                border: 1px solid #bababa;
                padding: 1% 0;
                border-left: none;
                padding-right: 1.5%;
                .top-sub {
                    color: #ededed;
                    font-size: 14px;
                    font-weight: bold;   
                }
            }
        }
       
        .bak {
            width: 12%;
            margin-top: 1.5%;
            }
      .userp {
        width: 35%;
        text-align: right;
        margin-top:1.2% 
        
      }
      .logout {
        width: 25%;
        background-color: #0067fc;
        border-radius: 4px;
        margin-top: 0.7%;
        color: #fff;
        height: 40px;
      }
}
}