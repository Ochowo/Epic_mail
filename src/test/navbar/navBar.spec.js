import React from 'react';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import { Header } from '../../components/navbar/navBar';
import { mockStore } from '../../../mocks/mockConfig';

const store = mockStore({ auth: {} });
const logoutMock = {
  preventDefault: () => ({}),
  logoutUser: jest.fn(),
};

const props = {
  logoutUser: () => {},
  auth: {
    isAuthenticated: true,
  },
  store,
};
const setUp = () => {
  const component = shallow(<Header {...props} store={store} />);
  return component;
};
describe('Navbar component', () => {
  let component;
  beforeEach(() => {
    component = setUp();
    sinon.spy(Header.prototype, 'onLogoutClick');
  });
  afterEach(() => {
    Header.prototype.onLogoutClick.restore();
  });
  it('should render without errors', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render correctly', (done) => {
    expect(component.find('header').length).toBe(1);
    done();
  });
  it('ensure logout', (done) => {
    const logout = component.find('.logout');
    expect(logout.length).toBe(1);
    logout.simulate('click', logoutMock);
    done();
  });
});
