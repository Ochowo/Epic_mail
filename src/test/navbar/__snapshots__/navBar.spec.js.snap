// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should render without errors 1`] = `ShallowWrapper {}`;
